### PR TITLE
build: switch from npm to yarn

### PR DESCRIPTION
**Pull request by ocombe**
## PR Type
What kind of change does this PR introduce?

```
[x] Build related changes
[x] CI related changes
```

## What is the current behavior?
We use npm to install and lock our dependencies (with shrinkwrap)

## What is the new behavior?
We use yarn instead of npm

## Does this PR introduce a breaking change?
```
[x] No
```

### CODE REVIEW

**Comment by 4kochi** [comment]
Just out of curiosity. Why not just use npm v5? It should be almost as fast as yarn. Or is there another reason besides speed improvements?

**Comment by ocombe** [comment]
We still need to support old versions of node for some tests, which I believe that the new npm doesn't.
Also we will try to use yarn workspaces after this to share the node folder across all packages.

And yarn is still faster for now, see https://medium.com/@nikjohn/npm-5-yarn-killer-ba69737b24d0

**Comment by 4kochi** [comment]
Ok, thanks for the clarification.

File: packages/common/src/pipes/date_pipe.ts [hunk 1]
```diff
@@ -148,14 +148,13 @@ export class DatePipe implements PipeTransform {
...
-      * If we leave the '-' ("2015-01-01") and try to create a new Date("2015-01-01") the timeoffset
-      * is applied
-      * Note: ISO months are 0 for January, 1 for February, ...
-      */
+       * For ISO Strings without time the day, month and year must be extracted from the ISO String
+       * before Date creation to avoid time offset and errors in the new Date.
+       * If we only replace '-' with ',' in the ISO String ("2015,01,01"), and try to create a new
+       * date, some browsers (e.g. IE 9) will throw an invalid Date error
+       * If we leave the '-' ("2015-01-01") and try to create a new Date("2015-01-01") the
+       * timeoffset is applied Note: ISO months are 0 for January, 1 for February, ...
```
Comment (IgorMinar) [inline packages/common/src/pipes/date_pipe.ts]: bad line join

File: package.json [hunk 2]
```diff
@@ -17,7 +17,8 @@
     "url": "https://github.com/angular/angular.git"
   },
   "scripts": {
-    "postinstall": "node tools/npm/copy-npm-shrinkwrap && webdriver-manager update",
+    "preinstall": "node -e \"if(process.env.npm_execpath.indexOf('yarn') === -1) throw new Error('Please use Yarn instead of NPM to install dependencies. See: https://yarnpkg.com/lang/en/docs/install/')\"",
+    "postinstall": "webdriver-manager update",
```
[the remark this answers is not in the document]
Comment (ocombe) [reply package.json]: I wasn't sure if it would fail something with this here, I'll add it

File: yarn.lock.readme.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+- ensure that your `node_modules` directory is not stale by running `yarn install`
+
+
+To add a new dependency do the following: `yarn add <packagename> --dev`
+
+To update an existing dependency do the following: run `yarn upgrade <packagename>@<version|latest> --dev`
+or `yarn upgrade <packagename> --dev` to update to the latest version that matches version constraint
+in `package.json`
+
+To Remove an existing dependency do the following: run `yarn remove <packagename>@<version|latest>`
```
Comment (IgorMinar) [inline yarn.lock.readme.md]: I don't think you need to specify the `version` in this case

File: docs/DEVELOPER.md [hunk 4]
```diff
@@ -56,7 +58,7 @@ Next, install the JavaScript modules needed to build and test Angular:
 
 ```shell
 # Install Angular project dependencies (package.json)
-npm install
+yarn install
 ```
 
 **Optional**: In this document, we make use of project local `npm` package scripts and binaries
```
[the remark this answers is not in the document]
Comment (ocombe) [reply docs/DEVELOPER.md]: done

File: scripts/ci/env.sh [hunk 5]
```diff
@@ -37,7 +37,7 @@ fi
 setEnvVar NODE_VERSION 6.9.5
 setEnvVar NPM_VERSION 3.10.7 # do not upgrade to >3.10.8 unless https://github.com/npm/npm/issues/14042 is resolved
```
Comment (IgorMinar) [inline scripts/ci/env.sh]: remove this line

**Review by IgorMinar — APPROVED** [review]

**Comment by alexeagle** [comment]
fyi on error-reporting: I had node 4 and the `yarn install` fails with
```
yarn install v1.0.2
$ node -e "if(process.env.npm_execpath.indexOf('yarn') === -1) throw new Error('Please use Yarn instead of NPM to install dependencies. See: https://yarnpkg.com/lang/en/docs/install/')"
error Couldn't find the binary node -e "if(process.env.npm_execpath.indexOf('yarn') === -1) throw new Error('Please use Yarn instead of NPM to install dependencies. See: https://yarnpkg.com/lang/en/docs/install/')"
```
not sure if it's worth fixing (by checking node version first I suppose)

**Comment by ocombe** [comment]
The node engine in package.json is `"node": ">=6.9.5 <7.0.0",`, so we probably don't need to support node 4?

**Comment by alexeagle** [comment]
Right, not asking for support, I'm just pointing out the error isn't very nice.
Igor is making fun of me for running node 4 at all; I don't care if you change anything for that case.

File: packages/platform-server/integrationtest/run_tests.sh [hunk 6]
```diff
@@ -8,6 +8,6 @@ echo "#################################"
 echo "Running platform-server end to end tests"
 echo "#################################"
 
-npm install
+yarn install
 
 npm run test
```
Comment (gkalpak) [inline packages/platform-server/integrationtest/run_tests.sh]: Not `yarn test`? 😞 😛

Comment (ocombe) [reply packages/platform-server/integrationtest/run_tests.sh]: good catch, it'll be in my follow up PR, thanks

**Pull request closed without merging**